crates/solver/src/settlement_access_list.rs

```diff
@@ -0,0 +1,104 @@
+use ethcontract::{Address, H256};
+use reqwest::{
+    header::{HeaderMap, HeaderValue},
+    Client, Url,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize)]
+pub struct TenderlyRequest {
+    pub network_id: String,
+    pub block_number: u64,
+    pub from: Address,
+    pub input: String,
+    pub to: Address,
+    pub generate_access_list: bool,
+}
```

Comment:
To reduce the number of ways you can create a nonsensical `TenderlyRequest` one could change the `network_id` to be a `u64` and `input` to be a `Vec<u8>`.
To make writing tests with `Vec<u8>` somewhat ergonomic one would probably have to use `hex::decode()`.

Reply:
Changed `input` to be `Bytes`, since it will be needed for usage in settlement submission as bytes anyway.
I will keep the `network_id` as `String`, since node clients response for network id is a `String` and also tenderly api expects `String`, so I don't see any advantage of having to convert `String -> u64 -> String`, although It's obvious it should be u64.